crates/hdk/src/hdk.rs

```diff
@@ -0,0 +1,366 @@
+use crate::prelude::*;
+
+#[cfg(feature = "mock")]
+use mockall::*;
+
+pub const HDK_NOT_REGISTERED: &str = "HDK not registered";
+
+use once_cell::sync::Lazy;
+/// This is a cell so it can be set many times.
+/// Every test needs its own mock so each test needs to set it.
+// pub static mut HDK: std::cell::Cell<Box<dyn HdkT>> = std::cell::Cell::new(Box::new(ErrHdk));
+use parking_lot::RwLock;
+pub static HDK: Lazy<RwLock<Box<dyn HdkT>>> = Lazy::new(|| RwLock::new(Box::new(ErrHdk)));
```

Comment:
This is a little weird to me as wasm is always single threaded.
Could we use a thread local and RefCell instead and save on the lock overhead?

Reply:
Actually now that I think about it how does the lazy work when wasm isn't persisted?
Maybe the intention is to be able to run this as regular rust for testing?
If so I still don't think you have multi threads (or at least it would be a bad idea to).
So the Lazy is going to be initialized on every zome call for the real use case which is not ideal.